From a7afcf789ef9e4641c8542971458b4964569ca35 Mon Sep 17 00:00:00 2001
From: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
Date: Tue, 17 Sep 2019 16:20:23 +0200
Subject: [PATCH 1/2] miral: Fix glob syntax in symbols.map

The glob syntax treats '[' as the start of a character range, so lld is
pretty reasonable in rejecting a single '[' in the symbol name.

'[[]' (a character range containing only '[') is the correct way to match
a '[' in glob syntax.
---
 src/miral/symbols.map | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/miral/symbols.map b/src/miral/symbols.map
index e04f5fbda2..881f3b7fa9 100644
--- a/src/miral/symbols.map
+++ b/src/miral/symbols.map
@@ -433,9 +433,9 @@ global:
     miral::WaylandExtensions::disable*;
     miral::WaylandExtensions::enable*;
 #    miral::WaylandExtensions::recommended*;
-    miral::WaylandExtensions::recommended[*;
+    miral::WaylandExtensions::recommended[[]*;
 #    miral::WaylandExtensions::supported*;
-    miral::WaylandExtensions::supported[*;
+    miral::WaylandExtensions::supported[[]*;
     miral::WaylandExtensions::zwlr_layer_shell_v1*;
     miral::WaylandExtensions::zxdg_output_manager_v1*;
     miral::WindowInfo::depth_layer*;

From 604500126c6497a69457f454740030cb3d7126e5 Mon Sep 17 00:00:00 2001
From: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
Date: Tue, 17 Sep 2019 16:46:06 +0200
Subject: [PATCH 2/2] miral: Also fix glob syntax in symbols generation script

---
 src/miral/regenerate-miral-symbols-map.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/miral/regenerate-miral-symbols-map.py b/src/miral/regenerate-miral-symbols-map.py
index 780312f7ab..fb6764bac5 100755
--- a/src/miral/regenerate-miral-symbols-map.py
+++ b/src/miral/regenerate-miral-symbols-map.py
@@ -516,9 +516,9 @@ def _report(publish, symbol):
     miral::WaylandExtensions::disable*;
     miral::WaylandExtensions::enable*;
 #    miral::WaylandExtensions::recommended*;
-    miral::WaylandExtensions::recommended[*;
+    miral::WaylandExtensions::recommended[[]*;
 #    miral::WaylandExtensions::supported*;
-    miral::WaylandExtensions::supported[*;
+    miral::WaylandExtensions::supported[[]*;
     miral::WaylandExtensions::zwlr_layer_shell_v1*;
     miral::WaylandExtensions::zxdg_output_manager_v1*;
     miral::WindowInfo::depth_layer*;
